src/common/definitions.h

```diff
@@ -127,6 +127,7 @@ IPtr<T> INew(Ptr<T> p) {
   return IPtr<T>(p);
 }
 
+/// enum class DeviceType
```

Comment:
This comment does not add much. Is it a documentation or is it required by doxygen to understand that it's an enum?

Reply:
This comment tells doxygen to get this enumeration documented (make link). I will add more details here.